service/history/mutableStateBuilder.go

```diff
@@ -1921,6 +1931,15 @@ func (e *mutableStateBuilder) addBinaryCheckSumIfNotExists(
 	exeInfo.AutoResetPoints = &workflow.ResetPoints{
 		Points: currResetPoints,
 	}
+	bytes, err := json.Marshal(recentBinaryChecksums)
+	if err != nil {
+		e.logger.Error(fmt.Sprintf("unable to serialize recent binary checksums %v", recentBinaryChecksums))
+	}
+	if exeInfo.SearchAttributes == nil {
+		exeInfo.SearchAttributes = make(map[string][]byte)
+	}
+	exeInfo.SearchAttributes[definition.BinaryChecksums] = bytes
+	e.AddTransferTasks(&persistence.UpsertWorkflowSearchAttributesTask{})
```

Comment:
Looks like it only works in active cluster in normal case. A few things need to look at:
1. Looks like AddTransferTasks should be called in taskGenerator so that it can obtain the version(for XDC), other wise it won't work with XDC.
2. Make sure that after reset, the search attrs are still correct. I believe you don't have to do anything because it's already the problem bowei has solved. But you need to verify.

Reply:
plz use the mutable state task generator

Reply:
Done